core/src/main/scala/kafka/tools/GetOffsetShell.scala

```diff
@@ -72,10 +73,10 @@ object GetOffsetShell {
                            .ofType(classOf[String])
     val timeOpt = parser.accepts("time", "timestamp of the offsets before that. [Note: No offset is returned, if the timestamp greater than recently committed record timestamp is given.]")
                            .withRequiredArg
-                           .describedAs("timestamp/-1(latest)/-2(earliest)")
+                           .describedAs("timestamp/-1(latest)/-2(earliest)/-3(max timestamp)")
                            .ofType(classOf[java.lang.Long])
                            .defaultsTo(-1L)
-    val commandConfigOpt = parser.accepts("command-config", s"Property file containing configs to be passed to Consumer Client.")
+    val commandConfigOpt = parser.accepts("command-config", s"Property file containing configs to be passed to Admin Client.")
```

Comment:
@dengziming , I think it is OK to extend the KIP-734 to support max timestamp in `GetOffsetShell`. And, it's good. But I don't think changing the `commandConfigOpt` from `Consumer client` into `Admin client` doesn't need KIP to address it. It'll cause backward compatibility issue because some some consumer configs won't work in admin config obviously. 

If the change from using `Consumer` into `AdminClient` is necessary (I think it's necessary because KIP-734 added the max timestamp support in `AdminClient`), I think you should have a KIP, to let everyone knows this is the change (and why we need it) that will break backward compatibility.

@dajac , what do you think?

Reply:
@showuon , Thank you for your remainders, I inspected `AdminClientConfig` carefully and I found the only config not presented in `ConsumerConfig` is `RETRIES_CONFIG`, and those consumer configs which can't work with admin config will be ignored. So, fortunately, this change is backward compatible thanks to the good design of `KafkaConfig`.

`RETRIES_CONFIG` is a new config for `GetOffsetShell`, but I don't think we would need a KIP for this change though, It's better to add this as an addendum to KIP-734 since they are related.

Reply:
@dengziming @showuon Good point. Using an addendum to KIP-734 will likely be unnoticed in the mailing list so  having a small KIP might be preferable here. @dengziming Are you willing to do it?

Reply:
@dajac sure, I create [KIP-813](https://cwiki.apache.org/confluence/display/KAFKA/KIP-813%3A+Replace+KafkaConsumer+with+AdminClient+in+GetOffsetShell) to track this issue and started a discussion. @showuon PTAL too.